src/convert/mapEvent.js

```diff
@@ -49,17 +64,23 @@ const mapEvent = (event) => {
 
   const keys = event.changes.keys;
   const changes = Array.from(keys.entries(), convertMapOp);
-  
-  var baseOp = {
-      type: (keys.has('data') && keys.get('data').action === 'update' && !event.path.length)?'set_value':'set_node',
+  let mapOperations = [];
+
+  const isSetValueOperation = keys.has("data") && event.path.length === 0;
+  const isSetNodeOperation = keys.has("data") && event.path.length !== 1;
+  if (isSetValueOperation || isSetNodeOperation) {
+    var baseOp = {
+      type: isSetValueOperation ? "set_value" : "set_node",
       newProperties: {},
       properties: {},
       path: toSlatePath(event.path),
     };
-
+    mapOperations = [changes.reduce(combineMapOp, baseOp)];
```

Comment:
in the interest of readability/maintainability, consider lifting this block of code out into something analogous to the new `mapInsertNodeOperations` function, then calling that function here instead of inlining it?